src/st/sel/3410C.c

```diff
@@ -0,0 +1,539 @@
+                continue;
+            }
+            if (i == 15) {
+                prim15 = prim;
+            }
+            switch (prim->p1) {
+            case 0:
+                if (i == 0xF) {
+                    if (!(rand() & 0x7F)) {
+                        prim->p2 = 6;
+                        prim->tpage = 0x19;
+                        prim->p1 += 1;
+                    }
+                } else {
+                    if ((prim15->p1 >= 2U) ||
+                        ((prim15->p1 != 0) && ((u8)prim15->p2 < 3U))) {
+                        prim->p2 = (rand() & 3) + 6;
+                        prim->tpage = 0x19;
+                        prim->p1 += 1;
+                    }
+                }
+                break;
+            case 1:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p2 = 6;
+                    prim->tpage = 0x1A;
+                    prim->p1 += 1;
+                }
+                break;
+            case 2:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p2 = (rand() & 7) + 6;
+                    prim->tpage = 0x1B;
+                    prim->p1 += 1;
+                }
+                break;
+            case 3:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p2 = 5;
+                    prim->tpage = 0x1A;
+                    prim->p1 += 1;
+                }
+                break;
+            case 4:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p2 = 5;
+                    prim->tpage = 0x19;
+                    prim->p1 += 1;
+                }
+                break;
+            case 5:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p2 = 6;
+                    prim->tpage = 0x18;
+                    prim->p1 += 1;
+                }
+                break;
+            case 6:
+                if (!(--prim->p2 & 0xFF)) {
+                    if (i != 0x10) {
+                        prim->p2 = 6;
+                        prim->p1 += 1;
+                        if (rand() & 3) {
+                            prim->p1 = 1;
+                            prim->tpage = 0x19;
+                        }
+                    } else {
+                        prim->p1 = 0;
+                    }
+                }
+                break;
+            case 7:
+                if (!(--prim->p2 & 0xFF)) {
+                    prim->p1 = 0;
+                }
+                break;
+            }
+
+            if (i != 0x10) {
+                continue;
+            }
+            if (prim->tpage == 0x18) {
+                prim->u1 = 0x78;
+            } else {
+                prim->u1 = 0x70;
+            }
+        }
+        if (g_pads[0].tapped & 0x800) {
+            g_api.PlaySfx(0x63D);
+            D_8003C9A4++;
+        }
+        return;
+    case 3:
+        D_8003C9A4 = 6;
+        return;
+    case 4:
+        if (g_pads[0].tapped & 0x6000) {
```

Comment:
```
        if (g_pads[0].tapped & (PAD_DOWN | PAD_RIGHT)) {
```

This is for the file select / name change screen.

Reply:
Thanks! Knew I was forgetting something! All button enums are now in.